File: slides2-kt/src/main/java/io/github/ilaborie/slides2/kt/jvm/tools/QrCode.java
package io.github.ilaborie.slides2.kt.jvm.tools;

import static io.github.ilaborie.slides2.kt.jvm.tools.NativeUtils.loadLib;

public class QrCode {

    static {
        loadLib("slides2_qrcode");
    }

    public static native String qrCode(String data);

}

File: slides2-kt/src/main/kotlin/io/github/ilaborie/slides2/kt/engine/plugins/qrcode.kt
package io.github.ilaborie.slides2.kt.engine.plugins

import io.github.ilaborie.slides2.kt.dsl.ContainerBuilder
import io.github.ilaborie.slides2.kt.jvm.tools.Natives


fun ContainerBuilder.qrCode(data: String) {
    val qr = Natives.qrCode(data)
    html { "<pre class=\"qrcode\">$qr</pre>" }
}
